NPE in ContainerBeansBuildItem

Camel build fails with NPE with Quarkus master snapshot:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project camel-quarkus-avro-integration-test: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step org.apache.camel.quarkus.core.deployment.BuildProcessor$Core#containerBeans threw an exception: java.lang.NullPointerException
[ERROR]         at org.apache.camel.quarkus.core.deployment.ContainerBeansBuildItem$SimpleCamelBeanInfo.<init>(ContainerBeansBuildItem.java:55)
[ERROR]         at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
[ERROR]         at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[ERROR]         at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
[ERROR]         at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[ERROR]         at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
[ERROR]         at org.apache.camel.quarkus.core.deployment.ContainerBeansBuildItem.<init>(ContainerBeansBuildItem.java:37)
[ERROR]         at org.apache.camel.quarkus.core.deployment.BuildProcessor$Core.containerBeans(BuildProcessor.java:109)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:938)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:273)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :camel-quarkus-avro-integration-test
```

--- a/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/ContainerBeansBuildItem.java
+++ b/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/ContainerBeansBuildItem.java
@@ -34,8 +34,10 @@ public final class ContainerBeansBuildItem extends SimpleBuildItem {
     private final Set<DotName> classes;
 
     public ContainerBeansBuildItem(Collection<BeanInfo> beans) {
-        this.beans = beans.stream().map(SimpleCamelBeanInfo::new).collect(Collectors.toSet());
-        this.classes = beans.stream().map(BeanInfo::getImplClazz).map(ClassInfo::name).collect(Collectors.toSet());
+        this.beans = beans.stream()
+                .filter(bi -> bi.getImplClazz() != null).map(SimpleCamelBeanInfo::new).collect(Collectors.toSet());
+        this.classes = beans.stream()
+                .map(BeanInfo::getImplClazz).filter(Objects::nonNull).map(ClassInfo::name).collect(Collectors.toSet());
     }
 
     public Set<CamelBeanInfo> getBeans() {

--- a/integration-tests/core-main/src/main/java/org/apache/camel/quarkus/core/CamelRouteProducer.java
+++ b/integration-tests/core-main/src/main/java/org/apache/camel/quarkus/core/CamelRouteProducer.java
@@ -24,6 +24,15 @@ import org.apache.camel.builder.RouteBuilder;
 
 @ApplicationScoped
 public class CamelRouteProducer {
+    /*
+     * The BeanInfo#getImplClazz() returns null in case of a produce of primitives ao arrays which
+     * cause ContainerBeansBuildItem to fail. This producer method is here only to validate we handle
+     * such case.
+     */
+    @Produces
+    public String[] primitiveType() {
+        return new String[] {};
+    }
 
     @Produces
     RoutesBuilder producedRoute() {

